packages/rest/src/router/routing-table.ts

```diff
@@ -66,15 +66,26 @@ export class RoutingTable {
 
     debug('Registering Controller with API', spec);
 
-    for (const path in spec.paths) {
-      for (const verb in spec.paths[path]) {
-        const opSpec: OperationObject = spec.paths[path][verb];
-        const route = new ControllerRoute(verb, path, opSpec, controller);
+    const basePath = spec.basePath || '/';
+    for (const p in spec.paths) {
+      for (const verb in spec.paths[p]) {
+        const opSpec: OperationObject = spec.paths[p][verb];
+        const fullPath = RoutingTable.joinPath(basePath, p);
+        const route = new ControllerRoute(verb, fullPath, opSpec, controller);
         this.registerRoute(route);
       }
     }
   }
 
+  private static joinPath(basePath: string, path: string) {
+    const fullPath = [basePath, path]
+      .join('/') // Join by /
+      .replace(/(\/){2,}/g, '/') // Remove extra /
+      .replace(/\/$/, '') // Remove trailing /
+      .replace(/^(\/)?/, '/'); // Add leading /
+    return fullPath;
+  }
```

Comment:
Is there an external utility that will handle this for us rather than creating our own?

Reply:
If we're going to keep our own hand-rolled implementation, then this should not be private, and it should have unit tests to validate its behaviour.